TheHub/TheHub.WebApp/Controllers/MediaController.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using TheHub.Library.Interfaces;
+using TheHub.Library.Model;
+
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace TheHub.WebApp.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class MediaController : ControllerBase
+    {
+
+        private readonly IMediaRepo _mediaRepository;
+
+        public MediaController(IMediaRepo mediaRepository)
+        {
+            _mediaRepository = mediaRepository;
+        }
+        // GET: api/<MediaController>
+        [HttpGet]
+        public IEnumerable<string> Get()
+        {
+            return new string[] { "value1", "value2" };
+        }
+
+        // GET api/<MediaController>/5
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+
+        public IActionResult GetMediaById(int id)
+        {
+            var currentMedia = _mediaRepository.GetById(id);
+
+            if (currentMedia == null)
+            {
+                return NotFound()
+            }
+            else
+            {
+                return Ok(_mediaRepository.GetById(id));
+            }
+        }
+
+        // POST api/<MediaController>
+        [HttpPost("CreateMedia")]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        public IActionResult CreateMedia([FromBody] Media media)
+        {
+            _mediaRepository.Add(media);
+
+            var newMedia = _mediaRepository.GetByTitle(media.MediaName);
+
+            return CreatedAtAction(nameof(GetMediaById), new { id = newMedia.MediaId }, newMedia);
+        }
+
+        // PUT api/<MediaController>/5
+        [HttpPut("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public IActionResult UpdateMedia(int id, [FromBody] Media media)
+        {
+           
+            var currentMedia = _mediaRepository.GetById(id);
+            if (currentMedia == null)
+            {
+                return NotFound();
+            }
+            else
+            {
+                currentMedia.MediaTypeId = media.MediaTypeId;
+                currentMedia.GenreId = media.GenreId;
+                currentMedia.MediaName = media.MediaName;
+                currentMedia.Description = media.Description;
+                currentMedia.MediaUrl = media.MediaUrl;
+
+                _mediaRepository.Update(currentMedia);
```

Comment:
Update method should already do lines 77-81 just given the media variable

Reply:
Fixed